Patch merged upstream
http://git.suckless.org/ii/commit/?id=f79e2f09534d92a6fe4e062b06449a925fef1c41

uses pledge():
  - stdio rpath wpath cpath dpath: ii(1) will create directory
    structure and fifos on demand (when joining a new channel for
    example).

--- ii.c.orig	Mon Jun  6 00:46:07 2016
+++ ii.c	Mon Jun  6 01:47:57 2016
@@ -386,7 +386,7 @@ static void handle_server_output() {
 }
 
 static void run() {
-	Channel *c;
+	Channel *c, *n;
 	int r, maxfd;
 	fd_set rd;
 	struct timeval tv;
@@ -423,9 +423,11 @@ static void run() {
 			handle_server_output();
 			last_response = time(NULL);
 		}
-		for(c = channels; c; c = c->next)
+		for(c = channels; c; c = n) {
+			n = c->next;
 			if(FD_ISSET(c->fd, &rd))
 				handle_channels_input(c);
+		}
 	}
 }
 
@@ -455,6 +457,14 @@ int main(int argc, char *argv[]) {
 			default: usage(); break;
 		}
 	}
+
+#ifdef __OpenBSD__
+	if (pledge("stdio rpath wpath cpath dpath", NULL) == -1) {
+		fputs("ii: pledge\n", stderr);
+		exit(EXIT_FAILURE);
+	}
+#endif
+
 	if(!snprintf(path, sizeof(path), "%s/%s", prefix, host)) {
 		fputs("ii: path to irc directory too long\n", stderr);
 		exit(EXIT_FAILURE);
